### PR TITLE
Apply streaming fix to apipie-rails master

### PR DESCRIPTION
**Pull request by brett-novu**
`apipie-rails` prevents developers from streaming responses:

```ruby
begin
  response.headers['Content-Type'] = 'text/event-stream'
  response.stream.write "[\n"
  # ...
  response.stream.write "\n]\n"
ensure
  response.stream.close unless response.stream.closed?
end
```

This appears to be due to the middleware `::Apipie::Extractor::Recorder::Middleware`, which should only be enabled when configured to do so.

### CODE REVIEW

**Comment by brett-novu** [comment]
I have updated this PR to be based on the latest master.

**Comment by mathieujobin** [comment]
Thanks for updating this PR...

Can you add a test that demonstrates the issue that is fixed?

**Review by mathieujobin — COMMENTED** [review]
please see inline comments as well

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
  Changelog
 ===========
 
+[v0.7.1](https://github.com/Apipie/apipie-rails/tree/v0.5.17.1) (2020-02-06)
```
[the remark this answers is not in the document]
Comment (brett-novu) [reply CHANGELOG.md:4]: Oops...  I messed that up...  Sorry about that.

File: lib/apipie/extractor.rb [hunk 2]
```diff
@@ -15,15 +15,18 @@ class Apipie::Railtie
...
     end
-    app.middleware.use ::Apipie::Extractor::Recorder::Middleware
 
-    if Gem::Version.new(Rails.version) < Gem::Version.new('5.0.0')
-      ActionController::TestCase::Behavior.instance_eval do
-        prepend Apipie::Extractor::Recorder::FunctionalTestRecording
+    if Apipie.configuration.record
+      app.middleware.use ::Apipie::Extractor::Recorder::Middleware
+
+      if Gem::Version.new(Rails.version) < Gem::Version.new('5.0.0')
```
[the remark this answers is not in the document]
Comment (brett-novu) [reply lib/apipie/extractor.rb:22]: Of course.

File: CHANGELOG.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (brett-novu) [reply CHANGELOG.md:4]: How best to handle the change log for PRs?

[the remark this answers is not in the document]
Comment (brett-novu) [reply CHANGELOG.md:4]: I'll just remove my changes to the change log so it can be handled when tagging.

**Comment by brett-novu** [comment]
> Thanks for updating this PR...
> 
> Can you add a test that demonstrates the issue that is fixed?

I will try...  It may require mocking.

**Comment by brett-novu** [comment]
I have yet to add specs, but otherwise made corrections requested.

**Comment by mathieujobin** [comment]
I'm only catching this now. can you please resolve the git conflict.
would love to see a spec being added ;-)

**Review by mathieujobin — APPROVED** [review]
Suddenly this makes sense and removing the rails <5.0 is good

**Comment by mathieujobin** [comment]
Sorry @brett-novu , Can I ask you to resolv the conflict one more time?

**Comment by mathieujobin** [comment]
if I was allowed to make changes to your branch, I would fix the conflict myself. but I can't

**Comment by mathieujobin** [comment]
merged into master manually in 0629831dae0c3bbc02b6c2a1002b004ed86d71d6

**Pull request closed without merging**